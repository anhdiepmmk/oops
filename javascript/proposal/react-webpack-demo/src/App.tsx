export const App = ({ message }: { message?: string }) => {
  return (
    <h1 data-testid="heading-hello">{message ? message : "Hello world"}</h1>
  );
};
