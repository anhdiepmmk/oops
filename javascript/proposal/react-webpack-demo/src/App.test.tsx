/** @jest-environment jsdom */
import { render, screen } from "@testing-library/react";
import { App } from "./App";

describe("App.tsx", () => {
  describe("when message not provied", () => {
    test("should show hello world as a default message", () => {
      render(<App />);

      expect(screen.getByTestId("heading-hello")).toHaveTextContent(
        "Hello world"
      );
      expect(screen.getByTestId("heading-hello")).toBeInTheDocument();
    });

    test("should match snapshot", () => {
      const { asFragment } = render(<App />);

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("when provide a message", () => {
    test("should show message instead of Hello world", () => {
      render(<App message="hi mom" />);

      expect(screen.getByTestId("heading-hello")).toHaveTextContent("hi mom");
      expect(screen.getByTestId("heading-hello")).toBeInTheDocument();
    });

    test("should match snapshot", () => {
      const { asFragment } = render(<App message="hi mom" />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
