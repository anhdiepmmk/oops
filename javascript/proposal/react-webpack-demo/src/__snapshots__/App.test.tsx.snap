// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx when message not provied should match snapshot 1`] = `
<DocumentFragment>
  <h1
    data-testid="heading-hello"
  >
    Hello world
  </h1>
</DocumentFragment>
`;

exports[`App.tsx when provide a message should match snapshot 1`] = `
<DocumentFragment>
  <h1
    data-testid="heading-hello"
  >
    hi mom
  </h1>
</DocumentFragment>
`;
